pkg/limits/frontend/ring.go

```diff
@@ -119,62 +119,52 @@ type getAssignedPartitionsResponse struct {
 	Response *logproto.GetAssignedPartitionsResponse
 }
 
-func (g *RingStreamUsageGatherer) getPartitionConsumers(ctx context.Context, rs ring.ReplicationSet) (map[int32]string, error) {
-	// Initialize result maps
-	highestTimestamp := make(map[int32]int64)
-	assigned := make(map[int32]string)
-
+func (g *RingStreamUsageGatherer) getPartitionConsumers(ctx context.Context, instances []ring.InstanceDesc) (map[int32]string, error) {
 	errg, ctx := errgroup.WithContext(ctx)
-	responses := make([]getAssignedPartitionsResponse, len(rs.Instances))
-
-	for i, instance := range rs.Instances {
+	responses := make(chan getAssignedPartitionsResponse, len(instances))
+	for _, instance := range instances {
 		errg.Go(func() error {
-			errChan := make(chan error, 1)
-			defer close(errChan)
-
-			loaderFunc := ttlcache.LoaderFunc[string, logproto.GetAssignedPartitionsResponse](
```

Comment:
We remove `ttlcache` in a subsequent PR (https://github.com/grafana/loki/pull/17318).